### PR TITLE
aaron/APPEALS-10313

### PR DESCRIPTION
**Pull request by Aaron-Willis**
Resolves #APPEALS-10313

### Description
Added Logic that will update an appeal's state within the **Appeal States Table** whenever an IHP task is **cancelled**.  This logic will specifically update the value of the **ihp_task_pending** column from **True** to **False** whenever an **IHP Task** is **cancelled**.  Updated Appeal State Logic for Appeal Docketed, IHP Pending, and IHP Complete so that it is now independent of Notification calls, resulting in appeal states for these event types always being up to date within the appeal state table.

### Acceptance Criteria
- [x] Cancelling an **IHP Task** will trigger an update to the row on the **Appeal States Table** for the corresponding task associated to the appeal.
- [x] A new module exists within **app/models/prepend/va_notify/** named **ihp_task_cancelled.rb**
- [x] The **ihp_task_pending** module is **prepended** to areas of the application that trigger when an **IHP Task** is **cancelled**.

### Testing Plan
1. Go to https://vajira.max.gov/browse/APPEALS-11318

### Testing Execution
1. Go to https://vajira.max.gov/browse/APPEALS-11695

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -91,7 +91,7 @@
     t.boolean "appeal_docketed", default: false, null: false, comment: "When true, appeal has been docketed"
     t.bigint "appeal_id", null: false, comment: "AMA or Legacy Appeal ID"
     t.string "appeal_type", null: false, comment: "Appeal Type (Appeal or LegacyAppeal)"
-    t.datetime "created_at", null: false, comment: "Date and Time the record was inserted into the table"
```
Comment (jshields13) [inline db/schema.rb:94]: I want to keep the comments so make sure yours looks like this

File: db/schema.rb [hunk 2]
```diff
@@ -100,7 +100,7 @@
     t.boolean "privacy_act_complete", default: false, null: false, comment: "When true, appeal has a privacy act request completed"
     t.boolean "privacy_act_pending", default: false, null: false, comment: "When true, appeal has a privacy act request still open"
     t.boolean "scheduled_in_error", default: false, null: false, comment: "When true, hearing was scheduled in error and none scheduled"
-    t.datetime "updated_at", comment: "Date and time the record was last updated"
```
Comment (jshields13) [inline db/schema.rb:103]: Keep these comments

**Review by jshields13 — CHANGES_REQUESTED** [review]

File: db/schema.rb (hunk 1 above)
Comment (Aaron-Willis) [reply db/schema.rb]: Reverted commit to add them back in.

File: db/schema.rb (hunk 2 above)
Comment (Aaron-Willis) [reply db/schema.rb]: Reverted commit to add comments back in

**Review by jshields13 — APPROVED** [review]

**Pull request merged**